device.go

```diff
@@ -0,0 +1,80 @@
+//
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package virtcontainers
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+const (
+	sysIOMMUPath = "/sys/kernel/iommu_groups"
+)
+
+// Device that must be available in the container.
+type Device struct {
+	// Full path to the device inside container.
+	Path string
+
+	// Type of device: c, b, u or p
```

Comment:
Can you expand this a bit to explain that `c` == character, etc?

Also, what happens if, say, `Type = foo`? I think we need validation on this field (we shouldn't blindly trust what we are provided in `config.json` because (a) security and (b) we don't know if we can support every conceivable value) and unit-tests for all scenarios (all valid values and a few invalid ones).